lib/adyen/rest/client.rb

```diff
@@ -97,14 +98,15 @@ def execute_http_request(request)
         end
       end
 
-      # The endpoint URI for this client.
+      # Set the endpoint URI for this client.
       # @return [URI] The endpoint to use for the environment.
-      def endpoint
-        @endpoint ||= URI(ENDPOINT % [environment])
+      def set_endpoint(resource)
+        @endpoint = URI(ENDPOINT % [environment, resource])
+        @endpoint
       end
 
       # @see Adyen::REST::Client#endpoint
-      ENDPOINT = 'https://pal-%s.adyen.com/pal/adapter/httppost'
+      ENDPOINT = 'https://pal-%s.adyen.com/pal/servlet/%s'
```

Comment:
Does this endpoint return the same result, but without the extra nesting level so we can drop the prefix stuff?